org.opentosca.container.core/src/main/java/org/opentosca/container/core/engine/ToscaEngine.java

```diff
@@ -439,14 +450,58 @@ public static Document getNodeTemplateProperties(TNodeTemplate nodeTemplate) {
 
     @Nullable
     public static Document getEntityTemplateProperties(TEntityTemplate template) {
-        return Optional.of(template)
-            .map(TEntityTemplate::getProperties)
-            // map via internal any to deal with HashMap property code in winery
-            .map(TEntityTemplate.Properties::getInternalAny)
-            .filter(p -> p instanceof Element)
-            .map(Element.class::cast)
-            .map(XMLHelper::fromRootNode)
-            .orElse(null);
+        DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+        documentBuilderFactory.setNamespaceAware(true);
+        try {
+            Document doc = documentBuilderFactory.newDocumentBuilder().newDocument();
+
+            if(template.getProperties() instanceof TEntityTemplate.WineryKVProperties){
+                TEntityTemplate.WineryKVProperties props = (TEntityTemplate.WineryKVProperties) template.getProperties();
+                Map<String, String> propMap = props.getKVProperties();
+
+                // So just that people understand:
+                /*
+                <Properties>
```

Comment:
Oh, what is this? Doesn't getKVProperties handle this correctly?

Reply:
Not really, as far is could test, however it could easily be fixed if elementName always returns Properties? Don't know ^^